content/training/R_Basics/Advanced_Functions.Rmd

```diff
@@ -196,10 +196,150 @@ This is often used to inspect which rows are unique to a data frame. An anti joi
 
+For further reading, see [this data visualization tutorial](https://r4ds.had.co.nz/data-visualisation.html) and [this gallery of visualizations and extensions](http://www.ggplot2-exts.org/gallery/) for inspiration.  
 
+### Core components  
 
+A `ggplot()` command is typically built using sub-commands linked by `+` symbols. The sub-commands are executed one-after-another to build the final plot. The core components to provide are:  
+
+* **a data set** - can be a standard data frame, or a spatial shapefile  
+* **coordinate system** - the x and y axes reflect the range of values within specified variables
+* **geom** - the geometric marks to visualize the data (e.g. points, lines, polygons, etc)  
+* **aesthetics** - linking attributes such as color and size of geoms to variables in the data set  
+
+The first command `ggplot()` creates an empty coordinate system to which layers of visualization and nuance are added.  
+* Its first argument specifies the data set to use.  
+* Its `aes()` argument establishes the variables to be mapped to the x and y axes.  
+
+Secondly, the geom is specified, indicating whether the data should be displayed as points, lines, bars, a map, etc. Without a geom, the data will not be visualized and the coordinate plane will be empty.  
+
+Other optional sub-commands can address the scale, labels, aesthetic themes and colors, etc. Note there are other ways to structure a ggplot() command, but the way described above is used in the template. When making a graphic, it often helps to look at examples online and mimic the code that produced them.  
+
+![](/images/ggplot.png)
+{{% notice tip %}}
+A command beginning with `ggplot()` will print the visualization but not save it. Using the assignment operator `<-` in the first line will assign the visualization to an R object and save it for later use. Print it by running a command of just the object name.
+{{% /notice %}}  
+
+### Geospatial mapping  
+
+In the Place section of the template, `ggplot()` is used to create geospatial maps of attack rates and other indicators. In these cases, the coordinate system reflects latitude/longitude, and `ggplot()` uses `geom_sf()` to accept and display points, polygons, or lines shapefiles (geom_sf is shorthand for "spatial file").
+
+
+### Aesthetic themes and labels  
+
+In the outbreak templates, detailed aesthetics parameters are managed by creating "theme" objects that can be referenced multiple in multiple `ggplot()` commands.  
+
+Below, the object `epicurve_theme` is assigned certain `ggplot()` aesthetic parameters such as x-axis text angle, absence of legend title, and grid-line color. This object is then referenced in `ggplot()` commands throughout the template.  
```

Comment:
I would avoid using aesthetic here since it has a different meaning for ggplot2

```suggestion
Below, the object `epicurve_theme` is assigned certain superficial plot parameters that are not associated with the data such as x-axis text angle, absence of legend title, and grid-line color. This object is then referenced in `ggplot()` commands throughout the template.  
```